website/tutorial/basic/01-project-setup.mdx

```diff
@@ -0,0 +1,130 @@
+---
+id: 01-project-setup
+title: Project Setup
+sidebar_label: Project Setup
+---
+
+In this section, you will learn how to create a basic NodeJS project with TypeScript.
+
+This step will cover the initial NodeJS setup required, basic configuration for TypeScript projects, and how to setup the development scripts.
+
+### Requirements
+
+- Have NodeJS installed on your operation system ([instructions](https://nodejs.org/en/download/package-manager/)). You can use any recent version (12/14/16).
+- Your favorite terminal configured (you are going to use it a lot!)
+- Run `node -v`, `npm -v`, `npx -v` in your terminal and make sure these commands are available for use.
+
+### Creating Node.js & TypeScript project
+
+This tutorial teaches you how to build a NodeJS project from scratch, so the first thing you need to do is create the
+directory that'll hold the files for your project:
+
+<Instruction>
+
+Open your terminal, navigate to a location of your choice, and run the following commands:
+
+```bash
+mkdir hackernews-node-ts
+cd hackernews-node-ts
+npm init -y
+```
+
+</Instruction>
+
+This creates a new directory called `hackernews-node-ts` and initializes it with a `package.json` file.
+
+The `package.json` is the configuration file for the Node.js app you're building. It lists all dependencies and other configuration options (such as _scripts_) needed for the app.
+
+To add TypeScript support for your NodeJS project, do the following:
+
+<Instruction>
+
+In your project directory, install the packages required to run a TypeScript project:
+
+```bash
+npm install --save-dev --exact typescript @types/node ts-node ts-node-dev cross-env
+```
+
+</Instruction>
+
+This will add the dependencies to your project, and will update the `package.json`.
+
+The command above will get you the following libraries installed in the project:
+
+- `typescript` is the basic TypeScript language support and compiler.
+- `@types/node` is a package that contains the basic TypeScript types for NodeJS environment.
+- `ts-node` and `ts-node-dev` are libraries that allows you to run `.ts` files directly, without a compilation step to JavaScript and automatically re-run your files after you apply file changes.
+- `cross-env` allows providing environment variables cross platform (windows environment variables are different 🤷).
+
+<Instruction>
+
+Now, to create a new TypeScript tools with the default configuration, run the following:
+
+```bash
+npx tsc --init
+```
+
+</Instruction>
+
+This will create a `tsconfig.json` file in your project. The [`tsconfig.json` is the TypeScript configuration file](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html), and it exists per-project. This is where you tell TypeScript compiler which files to compile and how to compile.
+
+<Instruction>
+
+To make it easier to run your project, replace the `"scripts"` section in your `package.json` file with the following:
+
+```json
+{
```

Comment:
highlight code changes
Check out:
1. https://www.tortilla.academy/Urigo/WhatsApp-Clone-Tutorial/master/next/step/6
2. https://github.com/Urigo/WhatsApp-Clone-Tutorial/blob/master%40next/.tortilla/manuals/views/step2.md#changed-srccomponentschatslistscreenchatslisttsx

Reply:
Created https://github.com/the-guild-org/the-guild-docs/issues/506 for tracking this